telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java

```diff
@@ -180,21 +185,41 @@ public TelescopeLayout(Context context, AttributeSet attrs, int defStyle) {
       requestCaptureFilter =
           new IntentFilter(RequestCaptureActivity.getResultBroadcastAction(context));
       requestCaptureReceiver = new BroadcastReceiver() {
-        @TargetApi(Build.VERSION_CODES.LOLLIPOP) @Override
+        @TargetApi(21) @Override
         public void onReceive(Context context, Intent intent) {
           unregisterRequestCaptureReceiver();
 
           int resultCode = intent.getIntExtra(RequestCaptureActivity.RESULT_EXTRA_CODE,
               Activity.RESULT_CANCELED);
           Intent data = intent.getParcelableExtra(RequestCaptureActivity.RESULT_EXTRA_DATA);
 
-          final MediaProjection mediaProjection =
-              projectionManager.getMediaProjection(resultCode, data);
-          if (mediaProjection == null) {
+          if (resultCode != Activity.RESULT_OK) {
             captureCanvasScreenshot();
             return;
           }
 
+          // The service needs to be running before we start the projection and there's no guarantee
+          // that it will have started once we return from startForegroundService. Rather than using
+          // binders, we'll just bounce the data through another broadcast from the service.
+          registerServiceStartedReceiver();
+          startForegroundService(resultCode, data);
```

Comment:
I'm checking the result code above so I can just drop it and hard-code it out the other side.